### ISSUE TITLE
Does a space model needs an "active spatial entity"?

### ISSUE THREAD

**Issue by hernanmd** [cormas/cormas#456]
The space model (CMSpaceModel) includes an instance variable named **activeSpatialEntity** which is almost unused.
Only one method references it (#warnCormasModelThat:changeTo:), which has no senders in the Pharo image.

In which case the model could use an active spatial entity?

**Comment by SergeStinckwich** [comment]
From class CMSpaceModel comment: 
activeSpatialEntity <ClassName> From the pov contextual menu, the spatial entity class that has been selected to be displayed

So apparently the class of a selected entities. I dunno what is the pov contextual menu actually?

**Comment by SergeStinckwich** [comment]
methods ``activeSpatialEntity`` and ``activeSpatialEntity: aClass`` have only one sender: ``warnCormasModelThat: oldName changeTo: newClass`` that is never called.

**Comment by SergeStinckwich** [comment]
Seems a good candidate for removal ;-)
What the others think?

**Comment by tomooda** [comment]
I'm not sure if we can remove them or not.
The Cormas code often uses ```perform:with:``` to invoke common methods.
I need the context behind the method to make a judgement.

**Comment by tomooda** [comment]
However, maybe we can just try to remove them and see how the models work. If broken, well can just recover the methods from the repository.... :-)

**Comment by SergeStinckwich** [comment]
Yes you are right. But even when you try to search in source code with Finder, there is no other mention about activeSpatialEntity

**Comment by hernanmd** [comment]
Maybe we could wait a comment from Pierre or a modeler because it could be used in other models?
(and in that case it would be better to include the methods as extensions?)

**Comment by pierbo** [comment]
#warnCormasModelThat:changeTo: is used to rename a class. But I think this is a mistake to use activeSpatialEntity into CMSpaceModel. So yes, it should be remove from this class

**Comment by SergeStinckwich** [comment]
> The activeSpatialEntity is the current focused spatial entity (a class) of a spatial grid (remember that you may have several spatial classes in Cormas = SpatialEntityElement, ie. Cell and the Aggregates). ActiveSpatialEntity is thus an attribute of SpaceInterface that is used for manipulation (#chooseEntityForManipulation) or for changing its PoV (#setDelimitor: integer of: aSpatialClass).

Not sure to understand, what you mean by the current focused spatial entity of a spatial grid?

**Comment by SergeStinckwich** [comment]
> #warnCormasModelThat:changeTo: is used to rename a class. But I think this is a mistake to use activeSpatialEntity into CMSpaceModel. So yes, it should be remove from this class

but nobody call this method ``warnCormasModelThat:changeTo:``. Who need to rename a class and why we need a method for that? There are already methods to rename Pharo classes.

**Comment by pierbo** [comment]
yes, #warnCormasModelThat:changeTo: can be removed for now, then we will check if a simple rename procedure will be ok (otherwise add some extra procedures)

**Comment by SergeStinckwich** [comment]
What do you want to remove? Classes?

**Comment by SergeStinckwich** [comment]
What do you want to remove? Classes?

**Comment by pierbo** [comment]
The method ##warnCormasModelThat:changeTo: can be removed, as well as the attribute #activeSpatialEntity of CMSpaceModel
